drivers/gpio/gpio_stm32.h

```diff
@@ -228,6 +228,10 @@ struct gpio_stm32_data {
 	const struct device *dev;
 	/* user ISR cb */
 	sys_slist_t cb;
+#ifdef CONFIG_PM_DEVICE
+	/* device power state */
+	uint32_t power_state;
```

Comment:
that is fine for now IMHO. But we should be able `struct device_pm` field `fms_state`. But I'm looking it right now and it is so confuse that may be better to keep this way right now.

Reply:
From what I can see in `pm_work_handler`, these seems like 2 different things.
But this could be reviewed